mycore-base/src/main/java/org/mycore/frontend/cli/MCRObjectCommands.java

```diff
@@ -1060,6 +1061,175 @@ public static void checkDerivatesInObjects(String baseId) {
         LOGGER.info("Check done for {} entries", Integer.toString(counter));
     }
 
+    /**
+     * Checks objects of the specified base ID for validity against their specified schemas.
+     * 
+     * @param baseId
+     *            the base part of a MCRObjectID e.g. DocPortal_document
+     */
+    @MCRCommand(
+        syntax = "validate objects of base {0}",
+        help = "Validates all objects of base {0} against their specified schema.",
+        order = 401)
+    public static void validateObjectsOfBase(String baseID) {
+        if (baseID == null || baseID.length() == 0) {
+            LOGGER.error("Base ID of objects required to check their schema validity.");
+            return;
+        }
+        MCRXMLMetadataManager mgr = MCRXMLMetadataManager.instance();
+        List<String> idList;
+        try {
+            idList = mgr.listIDsForBase(baseID);
+        } catch (ArrayIndexOutOfBoundsException e) {
+            LOGGER.error("Requested base ID {} not found in store.", baseID);
+            return;
+        }
+        int counter = 0;
+        int maxresults = idList.size();
+        if (maxresults == 0) {
+            LOGGER.warn("No IDs returned for base {}, nothing to check.", baseID);
+            return;
+        }
+        for (String objID : idList) {
```

Comment:
You could use a for loop with an index if you need a count.
for(int i counter=0;i<maxresults;counter++)